pkg/apis/gcp/validation/worker.go

```diff
@@ -0,0 +1,17 @@
+package validation
+
+import (
+	gcpaws "github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// ValidateWorkerConfig validates a WorkerConfig object.
+func ValidateWorkerConfig(workerConfig *gcpaws.WorkerConfig, volumeType *string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if *volumeType == "SCRATCH" && (*workerConfig.Volume.LocalSSDInterface != "NVME" && *workerConfig.Volume.LocalSSDInterface != "SCSI") {
+		allErrs = append(allErrs, field.NotSupported(field.NewPath("volume", "LocalSSDInterface"), *workerConfig.Volume.LocalSSDInterface, []string{"NVME", "SCSI"}))
+	}
```

Comment:
What about defining the valid ssd interfaces once only here?

```suggestion
var validVolumeLocalSSDInterfacesTypes = sets.NewString("NVME", "SCSI")

// ValidateWorkerConfig validates a WorkerConfig object.
func ValidateWorkerConfig(workerConfig *gcpaws.WorkerConfig, volumeType *string) field.ErrorList {
	allErrs := field.ErrorList{}

	if volumeType != nil && *volumeType == "SCRATCH" && workerConfig.Volume.LocalSSDInterface != nil && !validVolumeLocalSSDInterfacesTypes.Has(*workerConfig.Volume.LocalSSDInterface) {
		allErrs = append(allErrs, field.NotSupported(field.NewPath("volume", "localSSDInterface"), *workerConfig.Volume.LocalSSDInterface, validVolumeLocalSSDInterfacesTypes.List()))
	}
```

Does it make sense to throw an error when the volume type is not `SCRATCH` but the local ssd interface was proivded?

Reply:
Also, the `volumeType` must be named `SCRATCH`. Why is this? Actually, the volume type is originally defined in the `CloudProfile` here: https://github.com/gardener/gardener/blob/master/example/30-cloudprofile.yaml#L43-L49. Is there a reason why I can't call it differently?

Reply:
Because for `GCP` local ssd, the type of volume type should be `SCRATCH`.
And in `machine-controller-manager`, it also limit the type: https://github.com/gardener/machine-controller-manager/blob/master/pkg/apis/machine/validation/gcpmachineclass.go#L103